activesupport/lib/active_support/isolated_execution_state.rb

```diff
@@ -19,8 +19,8 @@ def isolation_level=(level)
 
         if level != isolation_level
           clear
-          singleton_class.alias_method(:current, "current_#{level}")
-          singleton_class.send(:private, :current)
+          singleton_class.alias_method(:current_object, "current_#{level}")
```

Comment:
Why `current_object` instead of keeping `current`?

Reply:
`current` points to the Hash that gets [`||=` initialized](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/isolated_execution_state.rb#L46) either on the Fiber or Thread, depending on the selected isolation menu. It wasn't clear how/whether I could use this Hash object as the cache key within ConnectionPool logic; the hash itself is mutable which disqualifies it for use as a cache key, and it would be weird to use it that way anyway; could I perhaps use the `.object_id` of that hash, i.e. `current.object_id`?

Reply:
> current points to the Hash ...

Sorry, my comment wasn't clear. I was suggesting to have `current` as the public method, the other one is private so the name doesn't matter that much. 

But now that I took a second look at the code:

> It wasn't clear how/whether I could use this Hash object as the cache key ...

Wouldn't `IsolatedExecutionState.unique_id` do the trick? The advantage being that you wouldn't hold a reference on thread / fiber, so that would avoid a bunch of leak scenarios. The only downside would be the few `owned by a different thread:` errors, but we could have the `Thread / Fiber #inspect` in that `unique_id`.

Reply:
> but we could have the Thread / Fiber #inspect in that unique_id.

Or have a WeakRef to the thread / fiber.

Reply:
I forgot to mention: even if we can use `current.object_id`, the reaper checks the `alive?` method of the connection owner in order to detect whether a Thread (or Fiber) has finished executing. I think this pushes in the direction of exposing `current_object`. i guess we could also consider introducing some intermediary class that the Fiber or Thread could generate that is consulted throughout the process of checking out a connection and checking if the owner is still alive, but it feels a bit overcomplicated.

Reply:
> the reaper checks the `alive?` method of the connection owner in order to detect whether a Thread (or Fiber) has finished executing. I think this pushes in the direction of exposing current_object

Hum, I missed that. Ok, so indeed we'll have to expose it. I'm really not fan of `current_object` though, could be `current`, `current_owner`, `current_executor` or something like that, but `current_object` is really not telling much.